From 101bee08b8441f279df657d4a45868de3f976c0c Mon Sep 17 00:00:00 2001
From: Anji J <anji.jagarlmudi@freescale.com>
Date: Thu, 26 May 2016 15:58:48 +0530
Subject: [PATCH 50/93] DNCPE-138 Rest external PHYs before driver starts

- Disable CONFIG_RESET_PHY_R, it is resetting PHYs after driver started.
- Call reset_phy() before driver initialization.
- Update bootcmd to stop pfe before kernel starts.
---
 board/freescale/ls1012ardb/eth.c |    1 +
 include/configs/ls1012a_common.h |    2 +-
 include/configs/ls1012ardb.h     |    1 -
 3 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/board/freescale/ls1012ardb/eth.c b/board/freescale/ls1012ardb/eth.c
index 24fdd83..638db95 100644
--- a/board/freescale/ls1012ardb/eth.c
+++ b/board/freescale/ls1012ardb/eth.c
@@ -41,6 +41,7 @@ int board_eth_init(bd_t *bis)
 	struct mdio_info mac1_mdio_info;
 	struct ccsr_scfg *scfg = (struct ccsr_scfg *)CONFIG_SYS_FSL_SCFG_ADDR;
 
+	reset_phy();
 
 	/*TODO Following config should be done for all boards, where is the right place to put this */
 	out_be32(&scfg->pfeasbcr, in_be32(&scfg->pfeasbcr) | SCFG_PPFEASBCR_AWCACHE0);
diff --git a/include/configs/ls1012a_common.h b/include/configs/ls1012a_common.h
index 57fc057..2895e7a 100644
--- a/include/configs/ls1012a_common.h
+++ b/include/configs/ls1012a_common.h
@@ -152,7 +152,7 @@
 #else
 #define CONFIG_BOOTARGS		"console=ttyS0,115200 root=/dev/ram0 " \
 				"earlycon=uart8250,mmio,0x21c0500"
-#define CONFIG_BOOTCOMMAND		"sf probe 0:0; sf read $kernel_load "\
+#define CONFIG_BOOTCOMMAND		"pfe stop; sf probe 0:0; sf read $kernel_load "\
 					"$kernel_start $kernel_size && "\
 					"bootm $kernel_load"
 #endif
diff --git a/include/configs/ls1012ardb.h b/include/configs/ls1012ardb.h
index 2eee84a..e94df74 100644
--- a/include/configs/ls1012ardb.h
+++ b/include/configs/ls1012ardb.h
@@ -24,7 +24,6 @@
 #define EMAC2_PHY_ADDR          0x1
 #define CONFIG_PHYLIB
 #define CONFIG_PHY_REALTEK
-#define CONFIG_RESET_PHY_R
 #endif
 
 /*
-- 
1.7.9.5

